src/queues/sqs/notification/test.rs

```diff
@@ -243,3 +244,127 @@ fn serialize_complaint_feedback_type() {
+                ],
+                "headersTruncated":false,
+                "headers":[ 
+                    { 
+                        "name":"From",
+                        "value":"\"Qux\" <qux@example.com>"
+                    }
+                ],
+                "commonHeaders":{
+                    "subject":"Complaint Subject"
+                }
+            },
+            "complaint": {
+                "userAgent":"AnyCompany Feedback Loop (V0.01)",
+                "complainedRecipients":[
+                    {
+                        "emailAddress":"baz@example.com"
+                    }
+                ],
+                "complaintFeedbackType":"abuse",
+                "arrivalDate":"2009-12-03T04:24:21.000-05:00",
+                "timestamp":"2012-05-25T14:59:38.623Z",
+                "feedbackId":"000001378603177f-18c07c78-fa81-4a58-9dd1-fedc3cb8f49a-000000"
+            }
+        },
+        {
+            "notificationType": "Delivery",
+            "mail": {
+                "timestamp":"2017-04-24T15:44:00 +0000",
+                "messageId":"003",
+                "source":"quuz@example.com",
+                "sourceArn": "arn:aws:ses:us-west-2:888888888888:identity/example.com",
+                "sourceIp": "127.0.3.0",
+                "sendingAccountId":"3456",
+                "destination":[
+                    "xyzzy@example.com"
+                ],
+                "headersTruncated":false,
+                "headers":[ 
+                    { 
+                        "name":"From",
+                        "value":"\"Quuz\" <quuz@example.com>"
+                    }],
+                "commonHeaders":{
+                    "subject":"Delivery Subject"
+                }
+            },
+            "delivery": {
+                "timestamp":"2014-05-28T22:41:01.184Z",
+                "processingTimeMillis":546,
+                "recipients":["xyzzy@example.com"],
+                "smtpResponse":"250 ok:  Message 64111812 accepted",
+                "reportingMTA":"a8-70.smtp-out.amazonses.com",
+                "remoteMtaIp":"127.0.2.0"
+            }
+        }]
+    "#).expect("JSON error");
+
+    assert_eq!(notifications[0].notification_type, NotificationType::Bounce);
+    assert_eq!(notifications[0].mail.timestamp, Utc.ymd(2016, 01, 27).and_hms(14, 05, 45));
+    assert_eq!(notifications[0].mail.message_id, "001");
+    assert_eq!(notifications[0].mail.source, Some(String::from("foo@example.com")));
+    assert_eq!(notifications[0].mail.source_arn, Some(String::from("arn:aws:ses:us-west-2:888888888888:identity/example.com")));
+    assert_eq!(notifications[0].mail.source_ip, Some(String::from("127.0.1.0")));
+    assert_eq!(notifications[0].mail.sending_account_id, Some(String::from("1234")));
+    assert_eq!(notifications[0].mail.destination, Some([String::from("bar@example.com")].to_vec()));
+    assert_eq!(notifications[0].mail.headers_truncated, Some(false));
+    // ...
```

Comment:
No need to assert every single property I think, just `notification_type` and any properties that are inserted as part of the bounce record should be sufficient. And an assertion for `notifications.len()`.

Actually, a bit stupidly on my part, I only just understood a couple of weeks ago that I didn't need to create all these properties when I was defining the `Notification` struct. For some reason, back then I assumed serde would fail to deserialize if I hadn't defined every single property. But of course, it doesn't, it just skips over any that aren't defined.

At some point we can go through and remove most of those properties because they're just getting in the way, making the code longer, and we're never going to use them.

Reply:
Ref: #189

Reply:
> No need to assert every single property I think

Perfect! That was my confusion :)

> At some point we can go through and remove most of those properties because they're just getting in the way, making the code longer, and we're never going to use them.

Great!